types.d.ts

```diff
@@ -0,0 +1,1003 @@
+
+        /**
+         * TODO: What is this used for?
+         */
+        loginAnonymous(state: mage.core.IState, options: any): Session;
+
+        /**
+         * Register a new user
+         */
+        register(state: mage.core.IState, username: string, password: string, options: any, callback: Function): void;
+    }
+
+    /**
+     * Core modules
+     *
+     * @type {MageCore}
+     * @memberOf Mage
+     */
+    core: {
+        State: { new(): mage.core.IState };
+
+        config: {
+            setTopLevelDefault(): void;
+
+            setDefaults(): void;
+
+            loadModuleConfig(): void;
+
+            get(): any;
+
+            getSource(): string;
+        }
+
+        sampler: {
+            set(path: string[], id: string, value: number): void;
+
+            inc(path: string[], id: string, increment: number): void;
+
+            sample(path: string[], id: string, value: number): void;
+
+            timedSample(path: string[], id: string, delta: number): void;
+        }
+    };
+
+    /**
+     * Logger module
+     *
+     * @type {Logger}
+     * @memberOf Mage
+     */
+    logger: Logger;
+
+    /**
+     * Session module
+     *
+     * @memberOf Mage
+     */
+    session: {
+        /**
+         * Register a new session
+         *
+         * @param {State} state
+         * @param {string} actorId
+         * @param {string} language
+         * @param {*} meta
+         */
+        register(state: mage.core.IState, actorId: string, language: string, meta: any): void;
+
+        /**
+         * TODO
+         *
+         * @param {State} state
+         * @param {string} fromActorId
+         * @param {string} toActorId
+         * @param {Function} callback
+         */
+        reassign(state: mage.core.IState, fromActorId: string, toActorId: string, callback: Function): void;
+    }
+
+    /**
+     * Time module
+     *
+     * @memberOf Mage
+     */
+    time: {
+        bend(offset: number, accelerationFactor: number, startAt: number): void;
+
+        getConfig(): TimeConfig;
+
+        msec(): number;
+
+        now(msecOut?: number): number;
+
+        sec(): number;
+
+        server(): number;
+
+        translate(timestamp: number, msecOut?: boolean): number;
+
+        unbend(): void;
+    }
+}
+
+declare var mage: Mage;
+
+declare namespace mage {
+    namespace archivist {
+        interface IVault {
+
+	    }
+    }
+
+    namespace core {
+        interface ITask {
+            setup(cb: (error?: Error) => void): void;
+            start(cb: (error?: Error) => void): void;
+            shutdown(cb: (error?: Error) => void): void;
+        }
+
+        interface IModule {
+            setup?(state: IState, cb: (error?: Error) => void): void;
+            start?(state: IState, cb: (error?: Error) => void): void;
+            shutdown?(state: IState, cb: (error?: Error) => void): void;
+        }
+
+        export interface IState {
+            /**
+             * Archivist instance
```

Comment:
Nevermind, I thought you made an "obvious" copy/paste error. Instead, I made a reading error. Problem is between keyboard and user.